dashboard/cron.py

```diff
@@ -90,22 +96,62 @@ def update_with_udw_file(request):
 
     return HttpResponse("loaded file info")
 
-# update FILE_ACCESS records from UDW
+# update FILE_ACCESS records from BigQuery
 def update_with_udw_access(request):
 
-    #select file record from UDW
-    file_access_sql = "select rr.COURSE_ID, rr.FILE_ID, rr.ACCESS_TIME, u.global_canvas_id as USER_ID " \
-                      "from (select r.course_id as COURSE_ID, " \
-                      "split_part(r.url, '/', 5) as FILE_ID, " \
-                      "r.timestamp as ACCESS_TIME, r.user_id AS user_id " \
-                      "from requests r " \
-                      "where r.course_id='" + UDW_COURSE_ID + "' " \
-                      "and r.url like '/courses/" + CANVAS_COURSE_ID + "/files/%') as rr, " \
-                      "user_dim u " \
-                      "where rr.user_id = u.id"
+    # Instantiates a client
+    bigquery_client = bigquery.Client()
+
+    datasets = list(bigquery_client.list_datasets())
+    project = bigquery_client.project
+
+    # list all datasets
+    if datasets:
+        logger.debug('Datasets in project {}:'.format(project))
+        for dataset in datasets:  # API request(s)
+            logger.debug('\t{}'.format(dataset.dataset_id))
+
+            # choose the right dataset
+            if ("learning_datasets" == dataset.dataset_id):
+                # list all tables
+                dataset_ref = bigquery_client.dataset(dataset.dataset_id)
+                tables = list(bigquery_client.list_tables(dataset_ref))  # API request(s)
+                for table in tables:
+                    if ("enriched_events" == table.table_id):
+                        logger.debug('\t{}'.format("found table"))
+
+                        # query to retrieve all file access events for one course
+                        query = 'select CAST(SUBSTR(JSON_EXTRACT_SCALAR(event, "$.object.id"), 35) AS STRING) AS FILE_ID, ' \
+                                'SUBSTR(JSON_EXTRACT_SCALAR(event, "$.membership.member.id"), 29) AS USER_ID, ' \
+                                'EVENT_TIME as ACCESS_TIME, ' \
+                                'SUBSTR(JSON_EXTRACT_SCALAR(event, "$.group.id"),31) AS COURSE_ID ' \
+                                'FROM learning_datasets.enriched_events ' \
+                                'where JSON_EXTRACT_SCALAR(event, "$.edApp.id") = \'http://umich.instructure.com/\' ' \
```

Comment:
We are querying off of `http://umich.instructure.com` this edApp but I will make sure they aren't other edApp that we are not missing data from. I know that caliper Live events have machine generated events that we would like to ignore, 
How about this `JSON_EXTRACT_SCALAR(event, "$.edApp.id") != http://cluster32-files.instructure.com/`?

Reply:
We are still waiting for the official Canvas live event document. Before that, we are not sure what is involved with the URL "http://cluster32-files.instructure.com". 

We are only dealing with 'http://umich.instructure.com/' edApp right now.